Incorrect PublicKeyAlgorithm.String() of x509.EdDilithium3 in parsed certificate

**Description:**

The `ParseCertificate` function is generating a certificate with incorrect `PublicKeyAlgorithm.String()` values.

**Steps to reproduce:**

1. Run the `TestCreateCert` test in the `app_test.go` file.
2. Observe the test failure with the error message indicating that the `PublicKeyAlgorithm.String()` values do not match the expected values.

**Expected result:**

The `ParseCertificate` function should generate a certificate with `PublicKeyAlgorithm.String()` set to `Ed448-Dilithium3`.

**Actual result:**

The `ParseCertificate` function generates a certificate with `PublicKeyAlgorithm.String()` set to `Ed25519-Dilithium3`.

**Code snippet:**

The relevant code snippet is:

https://github.com/cloudflare/circl/blob/75b28edc25ec569e6353a2b944b0b83d48a9c2e8/sign/eddilithium3/eddilithium.go#L31


```go
// PublicKey is the type of an EdDilithium3 public key.
type PublicKey struct {
	e ed448.PublicKey
	d mode3.PublicKey
}
```

app.go

```go
func CreateCert() ([]byte, error) {
	scheme := x509.CirclSchemeByPublicKeyAlgorithm(x509.EdDilithium3)
	pk, sk, err := scheme.GenerateKey()
	if err != nil {
		return nil, err
	}
	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
	serialNumber, _ := rand.Int(rand.Reader, serialNumberLimit)
	notBefore := time.Now()
	notAfter := notBefore.Add(365 * 24 * time.Hour)

	template := &x509.Certificate{
		SerialNumber: serialNumber,
		Subject: pkix.Name{
			Organization: []string{"Foo 1983"},
		},
		NotBefore:             notBefore,
		NotAfter:              notAfter,
		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
		BasicConstraintsValid: true,
	}

	cert, err := x509.CreateCertificate(rand.Reader, template, template, pk, sk)
	return cert, err
}
```

app_test.go

```go
func TestCreateCert(t *testing.T) {
    assert := assert.New(t)
    cert, _ := CreateCert()
    parsedCert, _ := x509.ParseCertificate(cert)
    assert.Equal("Foo 1983", parsedCert.Subject.Organization[0], "Organization should be Foo 1983")
    assert.Equal(x509.SignatureAlgorithm(17), parsedCert.SignatureAlgorithm, "SignatureAlgorithm should be x509.SignatureAlgorithm(17)")
    assert.Equal("Ed448-Dilithium3", parsedCert.SignatureAlgorithm.String(), "SignatureAlgorithm should be Ed448-Dilithium3")
    assert.Equal(x509.PublicKeyAlgorithm(5), parsedCert.PublicKeyAlgorithm, "PublicKeyAlgorithm should be x509.PublicKeyAlgorithm(5)")
    assert.Equal("Ed448-Dilithium3", parsedCert.PublicKeyAlgorithm.String(), "PublicKeyAlgorithm should be Ed448-Dilithium3")
}
```

go test

```
--- FAIL: TestCreateCert (0.00s)
                Error:          Not equal: 
                                expected: "Ed448-Dilithium3"
                                actual  : "Ed25519-Dilithium3"
                Test:           TestCreateCert
                Messages:       PublicKeyAlgorithm should be Ed448-Dilithium3
```

--- a/src/crypto/tls/common.go
+++ b/src/crypto/tls/common.go
@@ -190,6 +190,7 @@ const (
 	signatureRSAPSS
 	signatureECDSA
 	signatureEd25519
+	signatureEdDilithium2
 	signatureEdDilithium3
 )
 

--- a/src/crypto/tls/tls_cf.go
+++ b/src/crypto/tls/tls_cf.go
@@ -6,6 +6,7 @@ package tls
 import (
 	circlPki "github.com/cloudflare/circl/pki"
 	circlSign "github.com/cloudflare/circl/sign"
+	"github.com/cloudflare/circl/sign/eddilithium2"
 	"github.com/cloudflare/circl/sign/eddilithium3"
 )
 
@@ -20,6 +21,7 @@ var circlSchemes = [...]struct {
 	sigType uint8
 	scheme  circlSign.Scheme
 }{
+	{signatureEdDilithium2, eddilithium2.Scheme()},
 	{signatureEdDilithium3, eddilithium3.Scheme()},
 }
 

--- a/src/crypto/tls/tls_cf_circl_test.go
+++ b/src/crypto/tls/tls_cf_circl_test.go
@@ -13,11 +13,11 @@ import (
 	"time"
 
 	"github.com/cloudflare/circl/sign"
-	"github.com/cloudflare/circl/sign/eddilithium3"
+	"github.com/cloudflare/circl/sign/eddilithium2"
 )
 
 func TestPQSignatureSchemes(t *testing.T) {
-	pqCert := createPQCert(t, eddilithium3.Scheme())
+	pqCert := createPQCert(t, eddilithium2.Scheme())
 	rsaCert := Certificate{
 		Certificate: [][]byte{testRSACertificate},
 		PrivateKey:  testRSAPrivateKey,
@@ -47,13 +47,13 @@ func TestPQSignatureSchemes(t *testing.T) {
 			clientPQ:           true,
 			serverPQ:           false,
 			serverCerts:        pqAndRsaCerts,
-			expectedCertSigAlg: x509.PureEdDilithium3,
+			expectedCertSigAlg: x509.PureEdDilithium2,
 		},
 		{
 			clientPQ:           true,
 			serverPQ:           true,
 			serverCerts:        pqAndRsaCerts,
-			expectedCertSigAlg: x509.PureEdDilithium3,
+			expectedCertSigAlg: x509.PureEdDilithium2,
 		},
 		{
 			clientPQ:           true,

--- a/src/crypto/x509/x509.go
+++ b/src/crypto/x509/x509.go
@@ -241,6 +241,7 @@ const (
 	SHA384WithRSAPSS
 	SHA512WithRSAPSS
 	PureEd25519
+	PureEdDilithium2
 	PureEdDilithium3
 )
 
@@ -270,6 +271,7 @@ const (
 	DSA // Only supported for parsing.
 	ECDSA
 	Ed25519
+	EdDilithium2
 	EdDilithium3
 )
 
@@ -278,7 +280,8 @@ var publicKeyAlgoName = [...]string{
 	DSA:          "DSA",
 	ECDSA:        "ECDSA",
 	Ed25519:      "Ed25519",
-	EdDilithium3: "Ed25519-Dilithium3",
+	EdDilithium2: "Ed25519-Dilithium2",
+	EdDilithium3: "Ed448-Dilithium3",
 }
 
 func (algo PublicKeyAlgorithm) String() string {

--- a/src/crypto/x509/x509_cf.go
+++ b/src/crypto/x509/x509_cf.go
@@ -7,6 +7,7 @@ import (
 	circlPki "github.com/cloudflare/circl/pki"
 	circlSign "github.com/cloudflare/circl/sign"
 	"github.com/cloudflare/circl/sign/eddilithium3"
+	"github.com/cloudflare/circl/sign/eddilithium2"
 )
 
 // To add a signature scheme from Circl
@@ -21,6 +22,7 @@ var circlSchemes = [...]struct {
 	alg    PublicKeyAlgorithm
 	scheme circlSign.Scheme
 }{
+	{PureEdDilithium2, EdDilithium2, eddilithium2.Scheme()},
 	{PureEdDilithium3, EdDilithium3, eddilithium3.Scheme()},
 }
 

